src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -77,7 +77,9 @@ public void storeMessage(Message message) {
 
     return messages;
   }
-  
+  //fetch a list of users based on all of the messages stored in Dataset
+  //since set can not contain duplicates, we won't add a user multiple times
+  //from the messages posted by the same user
```

Comment:
Can you please take a stab at formatting this following the style outlined here in section 2.4: https://www.baeldung.com/javadoc. If we keep all of our documentation in that style there are tools that can automatically generate searchable documentation for our code :)

Also I would move the comment about why you chose to use a set to inside of the function. A good rule of thumb to know whether a detail should go in the function documentation or a comment inside the function is whether or not someone using the function needs to know about that detail. So for example with this function, you might have also implemented it a worse way where you do something like put all of the users into a hash map of "Username" --> # of messages. Then you could have iterated that hash map and returned just the keys. Both that (worse) method and the method you currently have do the same thing. As someone calling the function though I don't need to know how you chose to get the names of all of the users, I just want to know that it works.

Now, if I was coming through to change something inside this function, then the comment about why you chose to use a set could be helpful in understanding the reasoning behind the code.

This was really long 😬 but I hope that makes sense. If not then please let me know and I can try to come up with a better explanation!